neo4j/work/simple.py

```diff
@@ -226,12 +219,55 @@ def done(summary_metadata):
             on_success=run_metadata["on_success"],
             on_failure=run_metadata["on_failure"],
         )
+
+        def on_success_done(summary_metadata):
+            result_metadata.update(summary_metadata)
+            bookmark = result_metadata.get("bookmark")
+            if bookmark:
+                self._bookmarks_in = tuple([bookmark])
+                self._bookmark_out = bookmark
+
+        def on_records_extend_records(records):
+            result._records.extend(hydrant.hydrate_records(result.keys(), records))
+
+        if cx.PROTOCOL_VERSION == Bolt3.PROTOCOL_VERSION:
+            fetch_size = -1
```

Comment:
Agree with @2hdddg, the fetch size is a protocol concern. It is strange that parameters are chosen and handlers are installed outside of protocol. Ideally, we shall let the protocol decide how to do pull and discard.